clang_bind/cmake_frontend.py

```diff
@@ -0,0 +1,162 @@
+import os
+import json
+
+
+class Codemodel:
```

Comment:
Let's add some documentation for the purpose of the class and some reference to the CMake File API part